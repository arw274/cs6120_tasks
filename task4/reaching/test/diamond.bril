# Block 0 reaching definitions:
# Block 1 reaching definitions:
# Block 2 (ge) reaching definitions:
#   From block 0 : var one instr 0
#   From block 0 : var cond0 instr 1
#   From block 1 : var b instr 0
# Block 3 (zero) reaching definitions: 
#   From block 0 : var one instr 0
#   From block 0 : var cond0 instr 1
# Block 4 (end) reaching definitions: 
#   From block 0 : var one instr 0
#   From block 0 : var cond0 instr 1
#   From block 1 : var b instr 0
#   From block 2 : var input instr 1
#   From block 3 : var input instr 1

@main (input: int) {
    one: int = const 1;
    cond0: bool = ge input one;
    br cond0 .ge .zero;
    b: int = const one;
.ge:
    input: int = sub input one;
    jmp .end;
.zero:
    input: int = const input;
    jmp .end;
.end:
    ret;
}