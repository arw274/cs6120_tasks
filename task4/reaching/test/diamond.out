Function: main

Block 0: [{'dest': 'one', 'op': 'const', 'type': 'int', 'value': 1}, {'args': ['input', 'one'], 'dest': 'cond0', 'op': 'ge', 'type': 'bool'}, {'args': ['cond0'], 'labels': ['ge', 'zero'], 'op': 'br'}]
Block 1: [{'args': ['one'], 'dest': 'b', 'op': 'const', 'type': 'int'}]
Block 2: [{'label': 'ge'}, {'args': ['input', 'one'], 'dest': 'input', 'op': 'sub', 'type': 'int'}, {'labels': ['end'], 'op': 'jmp'}]
Block 3: [{'label': 'zero'}, {'args': ['input'], 'dest': 'input', 'op': 'const', 'type': 'int'}, {'args': ['end'], 'op': 'jmp.'}]
Block 4: [{'label': 'end'}, {'op': 'ret'}]
Block 0 reaching definitions:
Block 1 reaching definitions:
Block ge reaching definitions:
  From block 0 : var cond0 instr 1
  From block 1 : var b instr 0
  From block 0 : var one instr 0
Block zero reaching definitions:
  From block 0 : var cond0 instr 1
  From block 0 : var one instr 0
Block end reaching definitions:
  From block 1 : var b instr 0
  From block ge : var input instr 1
  From block 0 : var one instr 0
  From block 0 : var cond0 instr 1
  From block zero : var input instr 1

